CMakeLists.txt

```diff
@@ -544,7 +549,7 @@ add_subdirectory(iree/runtime)
 add_subdirectory(iree/schemas)
 add_subdirectory(iree/task)
 add_subdirectory(iree/testing)
-add_subdirectory(iree/test)
+add_subdirectory(iree/test EXCLUDE_FROM_ALL)
```

Comment:
This could use a comment for why this is excluded. Possibly link to https://groups.google.com/g/iree-discuss/c/EA9LyUsPpJk/m/w2btX6x1AwAJ (here or up where you `add_custom_target(iree-test-deps...`)

Or, is the `EXCLUDE_FROM_ALL` change a necessary part of this PR? Is it fully connected, or is this in an intermediate state? (The target selection and CMake refactoring seems far enough along to land without more work)

Reply:
Yes, this is ready to go. Added a comment.